File: backend/src/main/java/edu/netcracker/backend/model/TicketClass.java
package edu.netcracker.backend.model;

import edu.netcracker.backend.dao.annotations.Attribute;
import edu.netcracker.backend.dao.annotations.PrimaryKey;
import edu.netcracker.backend.dao.annotations.Table;
import lombok.*;

import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
@Table("ticket_class")
public class TicketClass {

    @PrimaryKey("class_id")
    @EqualsAndHashCode.Include
    private Long classId;

    @Attribute("class_name")
    private String className;

    @Attribute("trip_id")
    private Long tripId;

    @Attribute("ticket_price")
    private Integer ticketPrice;

    @Attribute("discount_id")
    private Long discountId;

    @Attribute("class_seats")
    private Integer classSeats;
  
    private Integer itemNumber;

    private Integer remainingSeats;

    private List<Service> services = new ArrayList<>();

    private Discount discount;
}

File: frontend/src/main/web/src/app/pages/admin/bundles/shared/model/ticket-class.ts
import {Service} from "./service";

export class TicketClass {
  constructor(
    public class_id : number,
    public class_name : string,
    public ticket_price : number,
    public item_number : number,
    public services: Service[]
  ) {}

  getTotalPrice() {
    let servicesPrice: number = this.services
    .map(service => service.getTotalPrice())
    .reduce(( acc, cur ) => acc + cur);
    return ((this.ticket_price || 0) * (this.item_number || 0) + servicesPrice);
  }
}
